aws-opentelemetry-distro/tests/amazon/opentelemetry/distro/test_otlp_sigv4_exporter.py

```diff
@@ -0,0 +1,301 @@
+CONTENT_TYPE = "application/x-protobuf"
+AUTHORIZATION_HEADER = "Authorization"
+X_AMZ_DATE_HEADER = "X-Amz-Date"
+X_AMZ_SECURITY_TOKEN_HEADER = "X-Amz-Security-Token"
+
+
+class TestAwsSigV4Exporter(TestCase):
+    def setUp(self):
+        self.testing_spans = [
+            self.create_span("test_span1", SpanKind.INTERNAL),
+            self.create_span("test_span2", SpanKind.SERVER),
+            self.create_span("test_span3", SpanKind.CLIENT),
+            self.create_span("test_span4", SpanKind.PRODUCER),
+            self.create_span("test_span5", SpanKind.CONSUMER),
+        ]
+
+        self.invalid_cw_otlp_tracing_endpoints = [
+            "https://xray.bad-region-1.amazonaws.com/v1/traces",
+            "https://xray.us-east-1.amaz.com/v1/traces",
+            "https://logs.us-east-1.amazonaws.com/v1/logs",
+            "https://test-endpoint123.com/test",
+            "xray.us-east-1.amazonaws.com/v1/traces",
+            "https://test-endpoint123.com/test https://xray.us-east-1.amazonaws.com/v1/traces",
+            "https://xray.us-east-1.amazonaws.com/v1/tracesssda",
+        ]
+
+        self.expected_auth_header = "AWS4-HMAC-SHA256 Credential=test_key/some_date/us-east-1/xray/aws4_request"
+        self.expected_auth_x_amz_date = "some_date"
+        self.expected_auth_security_token = "test_token"
+
+    @patch.dict(os.environ, {}, clear=True)
+    def test_sigv4_exporter_init_default(self):
+        """Tests that the default exporter is OTLP protobuf/http Span Exporter if no endpoint is set"""
+
+        exporter = OTLPAwsSigV4Exporter()
+        self.validate_exporter_extends_http_span_exporter(exporter, DEFAULT_ENDPOINT + DEFAULT_TRACES_EXPORT_PATH)
+        self.assertIsInstance(exporter._session, requests.Session)
+
+    @patch.dict(os.environ, {OTEL_EXPORTER_OTLP_TRACES_ENDPOINT: OTLP_CW_ENDPOINT}, clear=True)
+    @patch("botocore.session.Session")
+    def test_sigv4_exporter_init_valid_cw_otlp_endpoint(self, session_mock):
+        """Tests that the endpoint is validated and sets the aws_region but still uses the OTLP protobuf/http
+        Span Exporter exporter constructor behavior if a valid OTLP CloudWatch endpoint is set."""
+
+        mock_session = MagicMock()
+        session_mock.return_value = mock_session
+
+        mock_session.get_available_regions.return_value = ["us-east-1", "us-west-2"]
+        exporter = OTLPAwsSigV4Exporter(endpoint=OTLP_CW_ENDPOINT)
+
+        self.assertEqual(exporter._aws_region, "us-east-1")
+        self.validate_exporter_extends_http_span_exporter(exporter, OTLP_CW_ENDPOINT)
+
+        mock_session.get_available_regions.assert_called_once_with("xray")
+
+    @patch.dict("sys.modules", {"botocore": None})
+    def test_no_botocore_valid_xray_endpoint(self):
+        """Test that exporter defaults when using OTLP CW endpoint without botocore"""
+
+        exporter = OTLPAwsSigV4Exporter(endpoint=OTLP_CW_ENDPOINT)
+
+        self.assertIsNone(exporter._aws_region)
```

Comment:
We should be testing that the fallback exporter created is actually the `OTLPSpanExporter`